backport fix from #5712 (#5927)
cylc/cylc-flow: changes.d/fix.5712.md

> Fixed a bug preventing simulation of task retries if job submission unset.

Author: I've changed this to `submission retry delays`.

> Fixed a bug preventing simulation of task retries if `submission retry delays` unset.

Reviewer: ```suggestion
Fixed a bug preventing simulation of task retries when `submission retry delays` is not set.
```
